ibis/expr/types/arrays.py

```diff
@@ -1059,6 +1059,219 @@ def flatten(self) -> ir.ArrayValue:
+        """Return whether any element in the array is true.
+
+        Returns
+        -------
+        BooleanValue
+            Whether any element in the array is true
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     {
+        ...         "arr": [
+        ...             [True, False],
+        ...             [False],
+        ...             [True],
+        ...             [None, False],
+        ...             [None, True],
+        ...             [None],
+        ...             [],
+        ...             None,
+        ...         ]
+        ...     }
+        ... )
+        >>> t.mutate(x=t.arr.any())
+        ┏━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┓
+        ┃ arr                  ┃ x       ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━┩
+        │ array<boolean>       │ boolean │
+        ├──────────────────────┼─────────┤
+        │ [True, False]        │ True    │
+        │ [False]              │ False   │
+        │ [True]               │ True    │
+        │ [None, False]        │ False   │
+        │ [None, True]         │ True    │
+        │ [None]               │ NULL    │
+        │ []                   │ NULL    │
+        │ NULL                 │ NULL    │
+        └──────────────────────┴─────────┘
+        """
+        return ops.ArrayAny(self).to_expr()
+
+    def all(self) -> ir.BooleanValue:
+        """Return whether all elements in the array are true (ignoring nulls).
+
+        Returns
+        -------
+        BooleanValue
+            Whether all elements in the array are true (ignoring nulls).
+            Will be NULL if no elements are non-NULL.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     {
+        ...         "arr": [
+        ...             [True, False],
+        ...             [False],
+        ...             [True],
+        ...             [None, False],
+        ...             [None, True],
+        ...             [None],
+        ...             [],
+        ...             None,
+        ...         ]
+        ...     }
+        ... )
+        >>> t.mutate(x=t.arr.any())
+        ┏━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┓
+        ┃ arr                  ┃ x       ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━┩
+        │ array<boolean>       │ boolean │
+        ├──────────────────────┼─────────┤
+        │ [True, False]        │ False   │
+        │ [False]              │ False   │
+        │ [True]               │ True    │
+        │ [None, False]        │ False   │
+        │ [None, True]         │ True    │
+        │ [None]               │ NULL    │
+        │ []                   │ NULL    │
+        │ NULL                 │ NULL    │
+        └──────────────────────┴─────────┘
+        """
+        return ops.ArrayAll(self).to_expr()
+
+    def array_min(self) -> ir.NumericValue:
```

Comment:
```suggestion
    def min(self) -> ir.NumericValue:
```

We should name these consistently, and I don't think that this could ever mean anything other than a reduction within each array.

Reply:
Oh, gosh, I guess some of these are actually defined in some of our backends.

DuckDB seems to define `min` as elementwise min...

Ugh, I really do not like the naming scheme `array_*`.

Reply:
I think pluralizing the name of the singular reduction operation makes a lot of sense here and aligns well with the semantics of a collection.

- `.mins()`
- `.maxs()`
- `.means()`
- `.sums()`

I feel like this makes sense given the result shape of these operations is `Columnar`.

This also would never show up on non `Array` columns (Ibis has always intentionally worked like that), so there's much less chance of confusion.

Reply:
It is a little confusing to use this plural-sounding thing on scalars, eg ibis.literal([1, 2,3]).mins()?
I agree array_min is overly verbose, but I think I'd prefer that to confusing? Drawing a blank for alternatives...